src/docfx/build/toc/parser/MarkdownTocMarkup.cs

```diff
@@ -110,23 +116,36 @@ TableOfContentsInputItem GetItem(HeadingBlock block)
                 }
 
                 var link = block.Inline.FirstOrDefault(l => l is LinkInline);
-                var literalLink = block.Inline.FirstOrDefault(l => l is LiteralInline);
                 if (link != null && link is LinkInline linkInline)
                 {
                     if (!string.IsNullOrEmpty(linkInline.Url))
                         currentItem.Href = linkInline.Url;
                     if (!string.IsNullOrEmpty(linkInline.Title))
                         currentItem.DisplayName = linkInline.Title;
 
-                    literalLink = linkInline.FirstOrDefault(l => l is LiteralInline);
+                    currentItem.Name = GetName(linkInline.Select(l => l as LeafInline));
```

Comment:
nit: When `GetName` returns `null` here, the `LinkInline` code branch will mix with `LiteralInline` code branch